### PR TITLE
Add Aggregate by doc for tracing

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What is this feature?**

Adds documentation for metrics summary UI (Aggregate by), based upon https://github.com/grafana/grafana/pull/73201

Needed: 
* [x] Screenshots of the actual UI
* [x] Verify procedure for the doc
* [x] Benefits statement -- the importance of the "why" someone would use this

image

**Who is this feature for?**

Early adopters who will be using the metrics summary UI. 

**Which issue(s) does this PR fix?**:
Fixes: https://github.com/grafana/tempo-squad/issues/304

Related: https://github.com/grafana/grafana/pull/74239

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:114]:
```suggestion
1. In the **Aggregate by** row, select an option from the first drop-down box. For example, `span`.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (joey-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:114]:
```suggestion
1. In the **Aggregate by** row, select a scope from the first drop-down box. For example, `span`.
```

**Review by joey-grafana — COMMENTED** [review]
Thank for putting this together, Kim! Do you need anything else from me?

**Comment by joey-grafana** [comment]
> When users update the Aggregate by fields, do they have to press Run query to update the results? (My assumption was yes, based on other how we do other queries.)

Yes.
 
> Do results appear in the same table as the main query results or as a separate table?

They appear in a separate table.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [inline docs/sources/shared/datasources/tempo-search-traceql.md:107]:
```suggestion
Using **Aggregate by**, you can calculate RED metrics (total span count, percent erroring spans, and latency information) for spans of `kind=server` that match your filter criteria, grouped by one or more attributes. 
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [inline docs/sources/shared/datasources/tempo-search-traceql.md:104]:
```suggestion
[Enable the `metricsSummary` feature toggle](/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) in Grafana to use this feature. Your Grafana Tempo datasource must also point to a Tempo database with the [Metrics Summary API](https://grafana.com/docs/tempo/latest/api_docs/metrics-summary/) enabled. Contact Grafana Support to enable this feature in Grafana Cloud.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [inline docs/sources/shared/datasources/tempo-search-traceql.md:111]: I don't understand this right now, but maybe it will make more sense once I see the screenshot.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [inline docs/sources/shared/datasources/tempo-search-traceql.md:120]: I see we have this here but I was suggesting we bump it up since this API is off by default in Tempo and we want to let the user know ASAP that without this on, this won't work. 

Perhaps we say something like "If your Grafana Tempo datasource is connected to a Tempo database without the metrics summary API enabled, you will see an error message saying "loadprocessors not found" when you try to use the Aggregate By feature".

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:107]: @joey-grafana -- is the time selector wired up for this feature? I believe in the API we added support for time ranges, so if you say "give me these metrics over the last 10 minutes" we'll calculate those numbers over that exact range. 

I know we can only look back 1 hour in time. So what happens if I say "give me these metrics over the last 2 hours"? I would expect this to fail. What happens if I say "give me these metrics over a 1 hour period that happened 2 days ago"? I would also expect this to fail.

**Review by 09jvilla — CHANGES_REQUESTED** [review]
made suggestions!

**Comment by 09jvilla** [comment]
@knylander-grafana its not related to this PR directly but I'd argue that "Define tags" should just be a subheader of "Define filters". 

The filters at the top (service name, duration, span name) are just the things that we think it will be most common for folks to filter on, so we hardcode them (although as you call out, they are configurable). 

The "tags" section lets you add more filters, where you specify what tags you want to filter by, since it would be impractical to list every single tag as an optional filter.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:107]: @09jvilla, @mdisibio and I tested this today. The data changed depending upon the selections we chose in the time selector for past 5 minutes, 10 minutes, etc. I was running intro-to-mltp as my test ground.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:120]: Excellent point. I'll add that in.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:111]: Screenshot: 
image

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:107]: Great! 

Now we have to figure out what happens if the chosen time range includes times older than `(current_time - 1 hour`)!

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:111]: Got it. In this case, my suggestion would be 

`The RED metrics are calculated for every name and user agent combination found in your data.`

Suggest using `name` instead of `endpoint` since that's the exact thing shown in the selector.

Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:111]: What about this? 

```suggestion
When you use  **Aggregate by**,  the selections you make determine how the information is reported in the Table. Every combination that matches selections in your data is listed in the table. 
Each aggregate value, for example `intrinsic`:`name`, has a corresponding column in the results table.

For example, **names** matching `GET /:endpoint` with a **span.http.user_agent** of `k6/0.46` appeared in 31,466 spans. Instead of being listed by traces and associated spans, the query results are grouped by the the selections in **Aggregate by**.

The RED metrics are calculated for every endpoint and user agent combinations found in your data.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:108]:
```suggestion
This capability is based on the [metrics summary API](/docs/grafana-cloud/monitor-infrastructure/traces/metrics-summary-api/).
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:109]:
```suggestion

When you use  **Aggregate by**,  the selections you make determine how the information is reported in the Table. Every combination that matches selections in your data is listed in the table. 
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:124]:
```suggestion
```

**Comment by knylander-grafana** [comment]
> made suggestions!

Thank you for the suggestions! I rewrote some of the sections. See if they make more sense now.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:115]:
```suggestion
The RED metrics are calculated for every name and user agent combination found in your data.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (joey-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:129]: Note: the metrics summary only calculates summaries based on the last hour. So it does use the time range, but only up to an hours range.

**Review by joey-grafana — APPROVED** [review]

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:129]: Thank you! I will add a note. 
```suggestion
1. Optional: Select a **Time range** to expand or narrow the data set. Data is available for the last hour.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:108]:
```suggestion
This capability is based on the [metrics summary API](/docs/grafana-cloud/monitor-infrastructure/traces/metrics-summary-api/). 
Metrics summary only calculates summaries based on the last hour.
```

Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:108]: I tweaked this a bit.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:129]: @joey-grafana -- What happens if I select a time range outside of the last hour? Do I get an error? Or do we still return results but it is just over the last hour of data?

**Review by 09jvilla — APPROVED** [review]
I'm approving to unblock. I still left 1 question there but I think its less of a docs question and more of making sure that we're handling invalid time ranges properly. We should report back to the user some sort of error ('hey this is an invalid time range') if their selected time range includes any time >1 hour in the past.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (09jvilla) [reply docs/sources/shared/datasources/tempo-search-traceql.md:129]: Ideally we would return an error that says something along the lines of: "sorry you requested an invalid time range. try again with a time range sometime in the last 1 hour".

**Comment by knylander-grafana** [comment]
@09jvilla Thank you for the review and updating the comments. I've merged those changes in. I don't have merge rights for Grafana. I've asked on the doc channel for someone to merge. 

@joey-grafana  If it's Monday and this hasn't merged, would you merge it?

**Pull request merged**
